static/js/vue-cdr-access/src/components/browseDisplay.vue

```diff
@@ -147,46 +159,106 @@
             },
 
             updateUrl() {
-                let params = this.urlParams({ types: 'Work' });
+                let params = this.urlParams();
+
+                if (this.is_admin_unit) {
+                    this.default_work_type = 'Collection';
+                    delete params.format;
+                    delete params.browse_type;
+                } else if (this.containsFolderType(this.$route.query.types)) {
+                    this.default_work_type = 'Work,Folder'
+                }
+
+                params.types = this.default_work_type;
                 this.$router.push({ name: 'browseDisplay', query: params });
             },
 
-            retrieveData() {
+            updateParams() {
                 let params = this.urlParams();
 
                 if (this.is_collection || this.is_folder) {
                     params.types = 'Work';
                 }
 
-                let param_string = this.formatParamsString(params);
+                if (this.is_admin_unit) {
+                    params.types = 'Collection';
+
+                    delete params.format;
+                    delete params.browse_type;
+                    localStorage.removeItem('dcr-browse-type');
+
+                    this.browse_type = null;
+                    this.search_method = 'listJson';
+                } else if (this.browse_type === 'structure-display') {
+                    params.types = 'Work,Folder';
+                    delete params.format;
+                    this.search_method = 'listJson';
+                } else {
+                    this.search_method = 'searchJson';
+                }
+
+                if (this.containsFolderType(this.$route.query.types)) {
+                    params.types = 'Folder';
+                    this.search_method = 'listJson';
+                }
+
+                this.default_work_type = params.types;
+                return params;
+            },
+
+            retrieveData() {
+                let param_string = this.formatParamsString(this.updateParams());
                 this.uuid = location.pathname.split('/')[2];
 
-                get(`listJson/${this.uuid}${param_string}`).then((response) => {
+                get(`${this.search_method}/${this.uuid}${param_string}`).then((response) => {
                     this.record_count = response.data.resultCount;
                     this.record_list = response.data.metadata;
                     this.container_name = response.data.container.title;
                     this.container_metadata = response.data.container;
+                    this.is_page_loading = false;
                 }).catch(function (error) {
                     console.log(error);
                 });
+            },
+
+            findPageType() {
+                // Small hack to check outside of Vue controlled DOM to see what page type we're on
+                this.is_admin_unit = document.getElementById('is-admin-unit') !== null;
```

Comment:
I guess this is needed because we are getting the parent record in the same request as the children? I wonder if it'd be possible to pass the type of the parent in during initialization. Probably something we can think about later

Reply:
At this point it's used mainly to show the buttons or not and set the browse query